providers-sdk/v1/plugin/service.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package plugin
+
+import (
+	"errors"
+	"os"
+	"time"
+)
+
+type Service struct {
+	lastHeartbeat int64
+}
+
+var heartbeatRes HeartbeatRes
+
+func (s *Service) Heartbeat(req *HeartbeatReq) (*HeartbeatRes, error) {
+	if req.Interval == 0 {
+		return nil, errors.New("heartbeat failed, requested interval is 0")
+	}
+
+	now := time.Now().UnixNano()
+	s.lastHeartbeat = now
+
+	go func() {
```

Comment:
We should avoid launching goroutines like this as we risk leaking memory. it would be better to send a message to some sort of heartbeat component, which there is a single instance of that would be responsible for killing the process

Reply:
+1, let's improve that long-term